src/tests/functional/test_submissions.py

```diff
@@ -48,19 +52,37 @@ def _run_submission(self, competition_zip_path, submission_zip_path, expected_su
         # The submission table lists our submission!
         assert self.find('submission-manager table tbody tr:nth-of-type(1) td:nth-of-type(2)').text == submission_zip_path
 
+        # Check that md5 information was stored correctly
+        # TODO: Why in @$#! does this not work:
+        #   os.path.join("./src/tests/functional", submission_full_path)
```

Comment:
this is because `submission_full_path` is an absolute path, (it starts with a `/`) so `os.path.join()` can't join an absolute path on to the end of a different one. sooo, you could strip the leading `/` off the `submission_full_path` and it works:
```python
    def test_md5(self):
        submission_full_path = os.path.join(self.test_files_dir, 'submission_15.zip')
        f_string = md5(f"./src/tests/functional{submission_full_path}")
        join_path = md5(os.path.join('./src/tests/functional', submission_full_path[1:]))
        assert f_string == join_path
```
Not sure that is a _great_ solution, but at least we know why it doesn't work now...

Reply:
ughhhhhhhhhh, how have I not ran into this after so many years? Annoying. ty!

Pulling this comment and leaving in this:

```md5(f"./src/tests/functional{submission_full_path}")```